### ISSUE TITLE
[A11Y] Screen Reader doesn't announce what control gets focused

### ISSUE THREAD

**Issue by towebo** [dotnet/winforms#12180]
### Environment

VisualStudio.17.Preview/17.11.0-pre.5.0+35118.90

### .NET version

8.0

### Did this work in a previous version of Visual Studio and/or previous .NET release?

No, never in the out of process designer

### Issue description

When the out of process designer has focus and you try to move between the controls on the form screen readers aren't able to detect the focus change and are therefore quiet.
This makes it impossible to work with the designer in a sufficient way.
You can press F4 to focus the property editor and tab to the components combo to select the component you want to work with but that's extremely insufficient.

### Steps to reproduce

1. Start a screen reader like Narrator, NVDA or JAWS.
2. Start Visual Studio and create a new WinForms application targeting .NET 8.
3. Press Ctrl +Alt + L to focus the Solution Manager and navigate to the form Form1 and press Enter to open it.
4. Press Esc to ensure you're in the designer and at the top level, i e on the form.
5. Press Alt + V, X and press Enter to open the Toolbox.
6. Type Button and press Enter to add a button to the form.
7. Press Esc a couple of times to be really sure you're at the form level.
8. Press Tab to focus the button.

The screen reader doesn't announce that the button got focused.

**Comment by towebo** [comment]
One solution would be to make the out of process designer emit a message to screen readers that makes them announce the text passed.
When a control gets focus it should announce its name and type of control (try the old designer for .NET Framework did).
When a control is moved with the arrow keys it could announce the new position.
When a control is moved with the arrow keys and Ctrl is held down so it aligns with another control it could announce what edge it's aligned with.

There are probably other nice cases that can be added to.
